### PR TITLE
docs: improve prepare proposal comments

### PR DESCRIPTION
**Pull request by rootulp**
Fixes a few inconsistencies between the code and comments.

### CODE REVIEW

File: app/prepare_proposal.go [hunk 1]
```diff
@@ -15,15 +15,13 @@ import (
...
 // PrepareProposal fulfills the celestia-core version of the ABCI interface by
-// preparing the proposal block data. The square size is determined by first
-// estimating it via the size of the passed block data. Then, this method
-// generates the data root for the proposal block and passes it back to
-// tendermint via the BlockData. Panics indicate a developer error and should
-// immediately halt the node for visibility and so they can be quickly resolved.
+// preparing the proposal block data. This method generates the data root for
+// the proposal block and passes it back to tendermint via the BlockData. Panics
+// indicate a developer error and should immediately halt the node for
+// visibility and so they can be quickly resolved.
```
Comment (rootulp) [inline app/prepare_proposal.go:21]: Removed a sentence about square size estimating b/c that doesn't seem like an explicit step in this function

File: app/prepare_proposal.go [hunk 2]
```diff
@@ -32,10 +30,6 @@ func (app *App) PrepareProposal(req abci.RequestPrepareProposal) abci.ResponsePr
 			App: app.BaseApp.AppVersion(),
 		},
 	})
-	// filter out invalid transactions.
-	// TODO: we can remove all state independent checks from the ante handler here such as signature verification
-	// and only check the state dependent checks like fees and nonces as all these transactions have already
-	// passed CheckTx.
```
Comment (rootulp) [inline app/prepare_proposal.go:38]: I think this TODO is technically still possible but it seems like an optimization that we don't need to do. If we want to do it, proposal to create a GH issue to track and I can add the comment back.

File: app/prepare_proposal.go [hunk 3]
```diff
@@ -32,10 +30,6 @@ func (app *App) PrepareProposal(req abci.RequestPrepareProposal) abci.ResponsePr
 			App: app.BaseApp.AppVersion(),
 		},
 	})
-	// filter out invalid transactions.
```
Comment (rootulp) [inline app/prepare_proposal.go:35]: Moved this closer to the line that filters transactions.

File: app/prepare_proposal.go [hunk 4]
```diff
@@ -72,8 +68,6 @@ func (app *App) PrepareProposal(req abci.RequestPrepareProposal) abci.ResponsePr
 		panic(err)
 	}
 
-	// create the new data root by creating the data availability header (merkle
-	// roots of each row and col of the erasure data).
```
Comment (rootulp) [inline app/prepare_proposal.go:76]: The following lines don't actually create the data root. That only happens at the very end of the function body with `dah.Hash()`.

File: app/prepare_proposal.go (hunk 2 above)
Comment (cmwaters) [reply app/prepare_proposal.go:38]: Yeah can we create a GH issue to track this

**Review by cmwaters — APPROVED** [review]

File: app/prepare_proposal.go (hunk 2 above)
Comment (rootulp) [reply app/prepare_proposal.go:38]: https://github.com/celestiaorg/celestia-app/issues/3300

**Review by ninabarbakadze — APPROVED** [review]

**Pull request merged**